File: frontend/src/types/template.ts
export interface TemplateSectionField {
  key: string;
  label: string;
  enabled: boolean;
  order: number;
}

export interface TemplateHeader {
  bank_name: string;
  primary_color: string;
  secondary_color: string;
  show_logo: boolean;
  subtitle: string;
}

export interface TemplateFooter {
  text: string;
  show_page_numbers: boolean;
}

export interface TemplateConfig {
  header: TemplateHeader;
  sections: TemplateSectionField[];
  footer: TemplateFooter;
}

export interface ReportTemplate {
  id: string;
  lender_id: string;
  name: string;
  is_active: boolean;
  logo_path: string | null;
  config_json: TemplateConfig;
  created_at: string;
  updated_at: string;
}

export interface TemplateListResponse {
  templates: ReportTemplate[];
}

export interface TemplateFieldOption {
  key: string;
  label: string;
}

File: backend/app/schemas/template.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class TemplateSectionField(BaseModel):
    key: str
    label: str
    enabled: bool = True
    order: int


class TemplateHeader(BaseModel):
    bank_name: str
    primary_color: str = "#1a3b5c"
    secondary_color: str = "#f0f4f8"
    show_logo: bool = True
    subtitle: str = "Property Valuation Report"


class TemplateFooter(BaseModel):
    text: str = "Confidential - For internal use only"
    show_page_numbers: bool = True


class TemplateConfig(BaseModel):
    header: TemplateHeader
    sections: list[TemplateSectionField]
    footer: TemplateFooter


class TemplateCreate(BaseModel):
    name: str
    config_json: TemplateConfig


class TemplateUpdate(BaseModel):
    name: str | None = None
    config_json: TemplateConfig | None = None


class TemplateResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    lender_id: UUID
    name: str
    is_active: bool
    logo_path: str | None = None
    config_json: dict
    created_at: datetime
    updated_at: datetime


class TemplateListResponse(BaseModel):
    templates: list[TemplateResponse]


class TemplateFieldOption(BaseModel):
    key: str
    label: str
